pvlib/irradiance.py

```diff
@@ -2399,6 +2402,17 @@ def gti_dirint(poa_global, aoi, solar_zenith, solar_azimuth, times,
     return output
 
 
+def _gti_dirint_check_input(aoi):
+    r"""
+    Helper for gti_dirint
+
+    Raises Exceptions from input data that cause the algorithm to fail.
+    """
+    if (aoi >= 90).all():
```

Comment:
Consider `not (aoi < 90).any()` instead.  The difference is how `NaN` is treated.  With `(aoi >= 90).all()`, an array like `np.array([100, np.nan])` will pass the check, which is probably not what we want.